### ISSUE TITLE
False Positive: Declaration access can be weaker (when using traits)

### ISSUE THREAD

**Issue by rbairwell** [kalessil/phpinspectionsea#1384]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         |  Php Inspections (EA Ultimate) 3.0.2 |
| Language level | PHP7.3                     |

Current behaviour
--
In `TestClass` is reports that`Since the class is final, the member can be declared private` on `doThings` (`Declaration access can be weaker` inspection). However, changing it to `private` will then break the method signature and will cause static analysis systems to highlight it as an unused private method (okay, that's not a PHP-Inspections issue as such, but helps illustrate the knock on effect).
```php
<?php
declare(strict_types = 1);

trait TestTrait
{
    final public static function doStuff() : void
    {
        self::doThings();
    }

    abstract protected static function doThings() : void;
}

final class TestClass
{

    use TestTrait;

    protected static function doThings() : void
    {
        print 'Hello';
    }
}

TestClass::doStuff();
```

Expected behaviour
--
Not to suggest the change of protected to private.

It's worth noting that PHPIEA does NOT show the message if the class is extended instead of using a trait:
```
<?php
declare(strict_types = 1);

abstract class TestAbstract
{
    final public static function doStuff() : void
    {
        self::doThings();
    }

    abstract protected static function doThings() : void;
}

final class TestClass extends TestAbstract
{

    protected static function doThings() : void
    {
        print 'Hello';
    }
}

TestClass::doStuff();
```

Environment details
---
PhpStorm 2019.2
Build #PS-192.5728.108, built on July 24, 2019
Runtime version: 11.0.3+12-b304.10 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 972M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: GrepConsole, Key Promoter X, PHPUnit code coverage, String Manipulation, com.intellij.ideolog, net.seesharpsoft.intellij.plugins.csv, BashSupport, NEON support, com.kalessil.phpStorm.phpInspectionsUltimate, de.espend.idea.php.phpunit, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, izhangzhihao.rainbow.brackets, org.psliwa.idea.composer, ru.adelf.idea.dotenv

**Comment by kalessil** [comment]
Fixed!
